back/app/services/activities_service.rb

```diff
@@ -45,4 +47,20 @@ def create_invite_not_accepted_since_3_days_activities(now, last_time)
       LogActivityJob.perform_later(invite, 'not_accepted_since_3_days', nil, now.to_i)
     end
   end
+
+  def create_phase_ending_soon_activities(now)
```

Comment:
I think this is a good solution. Alternatively, I was thinking that perhaps we could delegate the generated activities to the participation methods, which would avoid the `voting?` conditional. This would be slightly less efficient though (we would have to go through all active participation contexts).

In this case, I think the `voting?` conditional is acceptable though.